crates/libcontainer/src/process/container_main_process.rs

```diff
@@ -276,4 +278,65 @@ mod tests {
         }
         Ok(())
     }
+
+    #[test]
+    #[serial]
+    fn test_sync_seccomp() -> Result<()> {
+        use std::io::Read;
+        use std::os::unix::io::IntoRawFd;
+        use std::os::unix::net::UnixListener;
+        use std::thread;
+        use utils::create_temp_dir;
+
+        let tmp_dir = create_temp_dir("test_sync_seccomp")?;
+        let scmp_file = std::fs::OpenOptions::new()
+            .write(true)
+            .create(true)
+            .open(tmp_dir.path().join("scmp_file"))?;
+        std::fs::OpenOptions::new()
+            .write(true)
+            .create(true)
+            .open(tmp_dir.path().join("socket_file.sock"))?;
+
+        let (mut main_sender, mut main_receiver) = channel::main_channel()?;
+        let (mut init_sender, mut init_receiver) = channel::init_channel()?;
+        let socket_path = tmp_dir.path().join("socket_file.sock");
+        let socket_path_seccomp_th = socket_path.clone();
+
+        let state = ContainerProcessState::default();
+        let want = serde_json::to_string(&state)?;
+        let seccomp_th = thread::spawn(move || {
+            sync_seccomp(
+                &LinuxSeccompBuilder::default()
+                    .listener_path(socket_path_seccomp_th)
+                    .syscalls(vec![LinuxSyscallBuilder::default()
+                        .action(LinuxSeccompAction::ScmpActNotify)
+                        .build()
+                        .unwrap()])
+                    .build()
+                    .unwrap(),
+                &state,
+                &mut init_sender,
+                &mut main_receiver,
+            )
+            .unwrap();
+        });
+
+        let fd = scmp_file.into_raw_fd();
+        let notify_th = thread::spawn(move || {
+            assert!(main_sender.seccomp_notify_request(fd).is_ok());
+            assert!(init_receiver.wait_for_seccomp_request_done().is_ok());
+        });
```

Comment:
Why do we need to use thread?

Reply:
yes you are right, we don't need that thread. 🙇🏻